python/lsst/afw/image/multiband.py

```diff
@@ -0,0 +1,1093 @@
+
+    def getNumPlanesUsed(self):
+        """Number of mask planes used
+
+        This is required to be the same for all of the single
+        band `Mask` objects.
+        """
+        return self._getNumPlanesUsed
+
+    def getMaskPlaneDict(self):
+        """Dictionary of Mask Plane bit values
+        """
+        return self._maskPlaneDict
+
+    def clearMaskPlaneDict(self):
+        """Reset the mask plane dictionary
+        """
+        mask = self._singles[0]
+        mask.clearMaskPlaneDict()
+        self._maskPlaneDict = mask.getMaskPlaneDict()
+
+    def addMaskPlane(self, name):
+        """Add a mask to the mask plane
+
+        Parameters
+        ----------
+        name: str
+            Name of the new mask plane
+        """
+        mask = self._singles[0]
+        idx = mask.addMaskPlane(name)
+        self._maskPlaneDict = mask.getMaskPlaneDict()
+        return idx
+
+    def removeMaskPlane(self, name):
+        """Remove a mask from the mask plane
+
+        Parameters
+        ----------
+        name: str
+            Name of the mask plane to remove
+        """
+        mask = self._singles[0]
+        mask.removeMaskPlane(name)
+        self._maskPlaneDict = mask.getMaskPlaneDict()
+
+    def clearAllMaskPlanes(self):
+        """Clear all the pixels
+        """
+        mask = self._singles[0]
+        mask.clearAllMaskPlanes()
+
+    def _getOtherMasks(self, others):
+        """Check if two masks can be combined
+
+        This method checks that `self` and `others`
+        have the same number of bands, or if
+        others is a single value, creates a list
+        to use for updating all of the `singles`.
+        """
+        if isinstance(others, MultibandMask):
+            if len(self.singles) != len(others.singles) or self.filters != others.filters:
+                msg = "Both `MultibandMask` objects must have the same number of bands to combine"
+                raise ValueError(msg)
+            result = [s for s in others.singles]
+        else:
+            result = [others]*len(self.singles)
+        return result
+
+    def __ior__(self, others):
+        _others = self._getOtherMasks(others)
+        singles = list(self.singles)
+        for n in range(len(self)):
+            singles[n] |= _others[n]
+        self._singles = tuple(singles)
+        return self
+
+    def __iand__(self, others):
+        _others = self._getOtherMasks(others)
+        singles = list(self.singles)
+        for n in range(len(self)):
+            singles[n] &= _others[n]
+        self._singles = tuple(singles)
+        return self
+
+    def __ixor__(self, others):
+        _others = self._getOtherMasks(others)
+        singles = list(self.singles)
+        for n in range(len(self)):
+            singles[n] ^= _others[n]
+        self._singles = tuple(singles)
+        return self
+
+    def __setitem__(self, args, value):
+        """Set a subset of the MultibandMask
+        """
```

Comment:
Because I saw the `set` method in the pybind11 wrapper for `Mask` but only just now realized that there is a similar method for `image`. I'll add this to `MultibandImage` as well.